### PR TITLE
Connected Components using DFS

### PR DESCRIPTION
**Pull request by ayaankhan98**
#### Description of Change
Added `connected_components.cpp` in `graphs`

finding the connected components in undirected graphs (implemented using DFS)

### CODE REVIEW

File: graph/connected_components.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+};
+
+void graph::addEdge(int u, int v)
+{
+  adj[u-1].push_back(v-1);
+  adj[v-1].push_back(u-1);
+}
+
+void graph::dfs()
```
Comment (cclauss) [inline graph/connected_components.cpp:29]: Please rename this function to `depth_first_search()` so the reader does not need to guess.

**Comment by cclauss** [comment]
cpplint is failing...
```
graph/connected_components.cpp:4:  Do not use namespace using-directives.  Use using-declarations instead.  [build/namespaces] [5]
graph/connected_components.cpp:7:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:8:  private: should be indented +1 space inside class graph  [whitespace/indent] [3]
graph/connected_components.cpp:13:  public: should be indented +1 space inside class graph  [whitespace/indent] [3]
graph/connected_components.cpp:14:  Missing space after ,  [whitespace/comma] [3]
graph/connected_components.cpp:14:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
graph/connected_components.cpp:15:  Missing space after ,  [whitespace/comma] [3]
graph/connected_components.cpp:17:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:24:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:30:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:32:  Missing space after ,  [whitespace/comma] [3]
graph/connected_components.cpp:34:  Missing spaces around <  [whitespace/operators] [3]
graph/connected_components.cpp:34:  Missing space before ( in for(  [whitespace/parens] [5]
graph/connected_components.cpp:34:  Missing space after ;  [whitespace/semicolon] [3]
graph/connected_components.cpp:35:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:37:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:38:  Missing space after ,  [whitespace/comma] [3]
graph/connected_components.cpp:45:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:47:  Missing space before ( in for(  [whitespace/parens] [5]
graph/connected_components.cpp:48:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:49:  Missing space before ( in if(  [whitespace/parens] [5]
graph/connected_components.cpp:50:  Missing space after ,  [whitespace/comma] [3]
graph/connected_components.cpp:55:  { should almost always be at the end of the previous line  [whitespace/braces] [4]
graph/connected_components.cpp:57:  Missing space after ,  [whitespace/comma] [3]
Total errors found: 26
```

File: graph/connected_components.cpp (hunk 1 above)
Comment (ayaankhan98) [reply graph/connected_components.cpp:29]: okay

**Comment by ayaankhan98** [comment]
trying to fix these automated tests.

**Comment by cclauss** [comment]
Could we have more self-documenting variable names?  There are a lot of single character variable names which make the code look old school.

**Comment by ayaankhan98** [comment]
okay sure, i will be imporving them in next pull request.

**Comment by ayaankhan98** [comment]
Nice

**Comment by deadshotsb** [comment]
You should consider using using Union-Find algorithm rather than using DFS

**Comment by cclauss** [comment]
Let’s have ___both___ algorithms and have a benchmark to ___prove___ that on is better than the other.

**Comment by deadshotsb** [comment]
@cclauss  Sounds great

**Comment by deadshotsb** [comment]
@cclauss  The code works fine am merging it and raising an issue about Union-Find Algorithm. @ayaankhan98 Good work.

**Comment by ayaankhan98** [comment]
@cclauss @deadshotsb the idea of benchmarking Union-Find and DFS sounds great, soon i will be implementing connected components using Union-Find Algorithm

**Pull request merged**